File: socks/socks4/socks4-scrape.js
const axios = require('axios');
const fs = require('fs');

const proxyUrl = 'https://raw.githubusercontent.com/hw630590/free-proxies/refs/heads/main/proxies/socks4/socks4.txt';

axios.get(proxyUrl)
  .then(response => {
    fs.writeFileSync('socks4_proxies.txt', response.data);
    console.log('SOCKS4 proxies saved to socks4_proxies.txt');
  })
  .catch(error => {
    console.error('Error fetching SOCKS4 proxies:', error.message);
  });

File: socks/socks4/socks4-scrape.py
import requests

proxy_url = "https://raw.githubusercontent.com/hw630590/free-proxies/refs/heads/main/proxies/socks4/socks4.txt"

try:
    response = requests.get(proxy_url)
    response.raise_for_status()
    with open("socks4_proxies.txt", "w") as file:
        file.write(response.text)
    print("SOCKS4 proxies saved to socks4_proxies.txt")
except requests.RequestException as e:
    print(f"Error fetching SOCKS4 proxies: {e}")
